The `idris2` build error doesn't appear related to `zsh` as best as I can tell:

```
make[1]: Entering directory '/private/tmp/nix-build-idris2-0.5.1.drv-0/source'
make -C libs/prelude IDRIS2=/private/tmp/nix-build-idris2-0.5.1.drv-0/source/build/exec/idris2 IDRIS2_INC_CGS=chez IDRIS2_PATH="/private/tmp/nix-build-idris2-0.5.1.drv-0/source/libs/prelude/build/ttc:/private/tmp/nix-build-idris2-0.5.1.drv-0/source/libs/base/build/ttc:/private/tmp/nix-build-idris2-0.5.1.drv-0/source/libs/contrib/build/ttc:/private/tmp/nix-build-idris2-0.5.1.drv-0/source/libs/network/build/ttc:/private/tmp/nix-build-idris2-0.5.1.drv-0/source/libs/test/build/ttc"
make[2]: Entering directory '/private/tmp/nix-build-idris2-0.5.1.drv-0/source/libs/prelude'
/private/tmp/nix-build-idris2-0.5.1.drv-0/source/build/exec/idris2 --build prelude.ipkg
Exception: (while loading libidris2_support.dylib) dlopen(libidris2_support.dylib, 0x0002): tried: '/nix/store/cz7smiama0px8cx9kvmz3c3rqbpw93cp-swift-corefoundation-unstable-2018-09-14/Library/Frameworks/libidris2_support.dylib' (no such file), '/nix/store/cz7smiama0px8cx9kvmz3c3rqbpw93cp-swift-corefoundation-unstable-2018-09-14/Library/Frameworks/libidris2_support.dylib' (no such file), 'libidris2_support.dylib' (no such file), '/usr/local/lib/libidris2_support.dylib' (no such file), '/usr/lib/libidris2_support.dylib' (no such file), '/private/tmp/nix-build-idris2-0.5.1.drv-0/source/libs/prelude/libidris2_support.dylib' (no such file), '/usr/local/lib/libidris2_support.dylib' (no such file), '/usr/lib/libidris2_support.dylib' (no such file)
make[2]: *** [Makefile:2: all] Error 255
make[2]: Leaving directory '/private/tmp/nix-build-idris2-0.5.1.drv-0/source/libs/prelude'
make[1]: *** [Makefile:76: prelude] Error 2
make[1]: Leaving directory '/private/tmp/nix-build-idris2-0.5.1.drv-0/source'
make: *** [Makefile:233: bootstrap] Error 2
```

@lovesegfault @veprbl

--- a/pkgs/shells/zsh/default.nix
+++ b/pkgs/shells/zsh/default.nix
@@ -37,7 +37,7 @@ stdenv.mkDerivation {
     })
   ];
 
-  nativeBuildInputs = [ autoreconfHook yodl perl groff util-linux texinfo ];
+  nativeBuildInputs = [ autoreconfHook perl groff util-linux texinfo ] ++ lib.optionals stdenv.isLinux [ yodl ];
 
   buildInputs = [ ncurses pcre ];
 
@@ -55,9 +55,11 @@ stdenv.mkDerivation {
   checkFlags = map (T: "TESTNUM=${T}") (lib.stringToCharacters "ABCDEVW");
 
   # XXX: think/discuss about this, also with respect to nixos vs nix-on-X
-  postInstall = ''
+  postInstall = lib.optionalString stdenv.isDarwin ''
+    make install.bin install.modules install.fns
+    '' + lib.optionalString stdenv.isLinux ''
     make install.info install.html
-
+    '' + ''
     mkdir -p $out/etc/
     cat > $out/etc/zprofile <<EOF
 if test -e /etc/NIXOS; then

